File: isma-intg-api/src/main/java/ru/nstu/isma/intg/api/methods/StageCalculator.java
package ru.nstu.isma.intg.api.methods;

import java.io.Serializable;

/**
 * @author Mariya Nasyrova
 * @since 01.09.14
 */
public abstract class StageCalculator implements Serializable {

    public abstract double yk(double step, double y, double f, double[] stages);

    public abstract double k(double step, double y, double f, double[] stages, double stagesY, double stagesF);

}

File: isma-intg-api/src/main/kotlin/ru/nstu/isma/intg/api/methods/StageCalculator.kt
package ru.nstu.isma.intg.api.methods

abstract class StageCalculator {
    abstract fun yk(step: Double, y: Double, f: Double, stages: DoubleArray): Double

    abstract fun k(step: Double, y: Double, f: Double, stages: DoubleArray, stagesY: Double, stagesF: Double): Double
}